engines/dfc_provider/app/services/dfc_provider/authorization_control.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Service used to authorize the user on DCF Provider API
+# It controls an OICD Access token and an enterprise.
+module DfcProvider
+  class AuthorizationControl
+    def initialize(access_token)
+      @access_token = access_token
+    end
+
+    def process
```

Comment:
looking at it in more detail, this method name can be easily improved, this is getting the user from the token.

I'd call it find_ofn_user

```ruby
def find_ofn_user
      decode_token
      Spree::User.where(email: @payload['email']).first
end
```